src/unstract/sdk/prompt.py

```diff
@@ -91,6 +107,12 @@ def _post_call(
             )
         return result
```

Comment:
Since we are returning result here, will there be a case where we could get "Used before defined" error for the variable result when we end up in any unhandled exception here? Should we add an additional handler to catch Exception in general too? @harini-venkataraman

Reply:
@gaya3-zipstack This will be handled by the variable initialized [here](https://github.com/Zipstack/unstract-sdk/blob/b2bb0786d28ea500f6fba5bceebcbf0626d74662/src/unstract/sdk/prompt.py#L64). So, this error will not be thrown in this case.